tests/test_train.py

```diff
@@ -40,3 +43,40 @@ def test_package_model(trained_rasa_model, parameters):
         assert parameters["prefix"] in file_name
 
     assert file_name.endswith(".tar.gz")
+
+
+@pytest.fixture
+def move_tempdir():
+    # Create a new *empty* tmp directory
+    shutil.rmtree(TEST_TEMP, ignore_errors=True)
+    os.mkdir(TEST_TEMP)
+    tempfile.tempdir = TEST_TEMP
+    yield
+    tempfile.tempdir = None
+    shutil.rmtree(TEST_TEMP)
+
+
+def test_train_temp_files(
+    move_tempdir,
+    default_domain_path,
+    default_stories_file,
+    default_stack_config,
+    default_nlu_data,
+):
+    train(
+        default_domain_path,
+        default_stack_config,
+        [default_stories_file, default_nlu_data],
+        force_training=True,
+    )
+
```

Comment:
There should also not be any temp files after the first training, shouldn't it? I would also add an assert here.

Reply:
Yes! Correct, but the `assert` at the end of the function already covers the case that temp files were created after the first training (because the second training only cleans up its own files). I can add another `assert` here for clarity.